lib/utils.js

```diff
@@ -50,7 +50,7 @@ exports.validateArg = function validateArg (args, config) {
         break;
 
       case Function:
-        if (toString.call(value) !== '[object Function]') {
+        if (typeof obj !== "undefined" && toString.call(value) !== '[object Function]') {
```

Comment:
validateArg is supposed to confirm that an argument is a particular type. It seems like a bad place to turn *all* function arguments into optionals system wide. There must be a better place for this, no?